xcdat/variable.py

```diff
@@ -0,0 +1,103 @@
+"""Variable module for Dataset data variable related functions."""
+import xarray as xr
+
+
+def open_variable(dataset: xr.Dataset, var_name: str) -> xr.DataArray:
+    """Opens a data variable of type DataArray with additional attributes.
+
+    This function calls DataArray accessors on the data variable to store
+    additional attributes, which are cached in the DataArray object.
+
+    Attributes include:
+
+    - Axis bounds information ("lat_bnds", "lon_bnds", "time_bnds")
+
+    Parameters
+    ----------
+    dataset : xr.Dataset
+        The parent Dataset.
+    var_name : str
+        The name of the variable to be opened.
+
+    Returns
+    -------
+    xr.DataArray
+        The variable with additional attributes.
+
+    Examples
+    --------
+    Import module:
+
+    >>> from xcdat.dataset import open_dataset
+    >>> from xcdat.variable import open_variable
+
+    Open a variable from a Dataset:
+
+    >>> ds = open_dataset("file_path") # Autogenerates bounds if missing
+    >>> tas = open_variable(ds, "tas")
+
+    Get lat bounds attribute:
+
+    >>> tas.bounds.lat
+
+    Get lon bounds attribute:
+
+    >>> tas.bounds.lon
+
+    Get time bounds attribute:
+
+    >>> tas.bounds.time
+    """
+    data_var = dataset[var_name].copy()
+
+    # Missing bounds in the Dataset is usually an indication that the Dataset
+    # was not open using xcdat's `open_dataset()` function.
+    # TODO: Check for "time_bnds"
+    missing_bnds = []
+    for attr in ["lat_bnds", "lon_bnds"]:
+        if dataset.get(attr) is None:
+            missing_bnds.append(attr)
+
+    if missing_bnds:
+        raise ValueError(
+            f"The Dataset is missing the following axis bounds: {', '.join(missing_bnds)}. "
+            "Make sure to open the Dataset using `xcdat.dataset.open_dataset()` to "
+            "autogenerate bounds before passing to this function."
+        )
+    else:
+        data_var.bounds._copy_from_parent(dataset)
```

Comment:
At the moment, `open_variable()` only checks for missing `lat_bnds` or `lon_bnds` in the Dataset because there is no function to generate missing `time_bnds`.